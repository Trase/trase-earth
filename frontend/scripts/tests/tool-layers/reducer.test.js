import {
  SET_NODE_ATTRIBUTES,
  GET_CONTEXT_LAYERS
  // GET_MAP_VECTOR_DATA,
  // SAVE_MAP_VIEW,
  // SELECT_BASEMAP,
  // SELECT_CONTEXTUAL_LAYERS,
  // TOGGLE_MAP,
  // TOGGLE_MAP_DIMENSION
} from 'react-components/tool/tool.actions';
import {
  TOOL_LAYERS__SET_MAP_DIMENSIONS,
  TOOL_LAYERS__SET_LINKED_GEOIDS
} from 'react-components/tool-layers/tool-layers.actions';
import reducer from 'react-components/tool-layers/tool-layers.reducer';
import initialState from 'react-components/tool-layers/tool-layers.initial-state';

test(SET_NODE_ATTRIBUTES, () => {
  const action = {
    type: SET_NODE_ATTRIBUTES
  };
  const newState = reducer(initialState, action);
  expect(newState).toMatchSnapshot();
});

test(TOOL_LAYERS__SET_MAP_DIMENSIONS, () => {
  const dimensions = [
    {
      groupId: 3,
      layerAttributeId: 11,
      type: 'quant',
      uid: 'quant11'
    }
  ];
  const dimensionGroups = [
    { id: 3, name: 'Agricultural indicators' },
    { id: 1, name: 'Environmental indicators' }
  ];
  const action = {
    type: TOOL_LAYERS__SET_MAP_DIMENSIONS,
    payload: { dimensions, dimensionGroups }
  };
  const newState = reducer(initialState, action);
  expect(newState).toMatchSnapshot();
});

describe(TOOL_LAYERS__SET_LINKED_GEOIDS, () => {
  it('sets in linkedGeoIds an array of the nodes geoIds', () => {
    const action = {
      type: TOOL_LAYERS__SET_LINKED_GEOIDS,
      payload: { nodes: [{ id: 1, name: 'Node', geoId: 'BR-123-23' }] }
    };
    const newState = reducer(initialState, action);
    expect(newState).toMatchSnapshot();
  });

  it('sets in linkedGeoIds an empty array if there are no nodes', () => {
    const action = {
      type: TOOL_LAYERS__SET_LINKED_GEOIDS,
      payload: { nodes: [] }
    };
    const newState = reducer(initialState, action);
    expect(newState).toMatchSnapshot();
  });
});

test(GET_CONTEXT_LAYERS, () => {
  const mapContextualLayers = [
    {
      id: 10,
      identifier: 'landcover',
      isDefault: false,
      title: 'Land cover'
    }
  ];
  const action = {
    type: GET_CONTEXT_LAYERS,
    payload: { mapContextualLayers }
  };
  const newState = reducer(initialState, action);
  expect(newState).toMatchSnapshot();
});
