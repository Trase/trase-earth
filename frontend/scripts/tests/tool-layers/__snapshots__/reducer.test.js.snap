// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET_CONTEXT_LAYERS 1`] = `
Object {
  "data": Object {
    "mapContextualLayers": Object {
      "10": Object {
        "id": 10,
        "identifier": "landcover",
        "isDefault": false,
        "title": "Land cover",
      },
    },
    "mapDimensions": Object {},
    "mapDimensionsGroups": Array [],
    "mapVectorData": null,
  },
  "highlightedNodeCoordinates": null,
  "isMapVisible": false,
  "linkedGeoIds": Array [],
  "mapLoading": false,
  "mapView": null,
  "selectedMapBasemap": null,
  "selectedMapContextualLayers": null,
  "selectedMapDimensions": null,
}
`;

exports[`SET_NODE_ATTRIBUTES 1`] = `
Object {
  "data": Object {
    "mapContextualLayers": Array [],
    "mapDimensions": Object {},
    "mapDimensionsGroups": Array [],
    "mapVectorData": null,
  },
  "highlightedNodeCoordinates": null,
  "isMapVisible": false,
  "linkedGeoIds": Array [],
  "mapLoading": false,
  "mapView": null,
  "selectedMapBasemap": null,
  "selectedMapContextualLayers": null,
  "selectedMapDimensions": null,
}
`;

exports[`TOOL_LAYERS__SET_LINKED_GEOIDS sets in linkedGeoIds an array of the nodes geoIds 1`] = `
Object {
  "data": Object {
    "mapContextualLayers": Array [],
    "mapDimensions": Object {},
    "mapDimensionsGroups": Array [],
    "mapVectorData": null,
  },
  "highlightedNodeCoordinates": null,
  "isMapVisible": false,
  "linkedGeoIds": Array [
    "BR-123-23",
  ],
  "mapLoading": false,
  "mapView": null,
  "selectedMapBasemap": null,
  "selectedMapContextualLayers": null,
  "selectedMapDimensions": null,
}
`;

exports[`TOOL_LAYERS__SET_LINKED_GEOIDS sets in linkedGeoIds an empty array if there are no nodes 1`] = `
Object {
  "data": Object {
    "mapContextualLayers": Array [],
    "mapDimensions": Object {},
    "mapDimensionsGroups": Array [],
    "mapVectorData": null,
  },
  "highlightedNodeCoordinates": null,
  "isMapVisible": false,
  "linkedGeoIds": Array [],
  "mapLoading": false,
  "mapView": null,
  "selectedMapBasemap": null,
  "selectedMapContextualLayers": null,
  "selectedMapDimensions": null,
}
`;

exports[`TOOL_LAYERS__SET_MAP_DIMENSIONS 1`] = `
Object {
  "data": Object {
    "mapContextualLayers": Array [],
    "mapDimensions": Object {
      "quant11": Object {
        "groupId": 3,
        "layerAttributeId": 11,
        "type": "quant",
        "uid": "quant11",
      },
    },
    "mapDimensionsGroups": Array [
      Object {
        "dimensions": Array [
          "quant11",
        ],
        "id": 3,
        "name": "Agricultural indicators",
      },
      Object {
        "dimensions": Array [],
        "id": 1,
        "name": "Environmental indicators",
      },
    ],
    "mapVectorData": null,
  },
  "highlightedNodeCoordinates": null,
  "isMapVisible": false,
  "linkedGeoIds": Array [],
  "mapLoading": false,
  "mapView": null,
  "selectedMapBasemap": null,
  "selectedMapContextualLayers": null,
  "selectedMapDimensions": null,
}
`;
